Build failure: python-prctl

### Steps To Reproduce

Steps to reproduce the behavior:
1. nix-build -A python39Packages.python-prctl
or
3. nix-build -A python311Packages.python-prctl

### Build log

```
Executing pytestCheckPhase
============================= test session starts ==============================
platform linux -- Python 3.9.16, pytest-7.2.1, pluggy-1.0.0
rootdir: /build/python-prctl-1.8.1
collected 0 items / 1 error                                                    

==================================== ERRORS ====================================
________________________ ERROR collecting test_prctl.py ________________________
test_prctl.py:32: in <module>
    sys.exit(1)
E   SystemExit: 1
------------------------------- Captured stderr --------------------------------
Please build the extension first, using ./setup.py build
=========================== short test summary info ============================
ERROR test_prctl.py - SystemExit: 1
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.06s ===============================
/nix/store/74l4x6m97bvry7ccxv51h952ayvg2j46-stdenv-linux/setup: line 1596: pop_var_context: head of shell_variables not a function context
error: builder for '/nix/store/20ymk4hq8mvp75n24h1528vh1h6x5mnq-python3.9-python-prctl-1.8.1.drv' failed with exit code 2;
       last 10 log lines:
       > test_prctl.py:32: in <module>
       >     sys.exit(1)
       > E   SystemExit: 1
       > ------------------------------- Captured stderr --------------------------------
       > Please build the extension first, using ./setup.py build
       > =========================== short test summary info ============================
       > ERROR test_prctl.py - SystemExit: 1
       > !!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
       > =============================== 1 error in 0.06s ===============================
       > /nix/store/74l4x6m97bvry7ccxv51h952ayvg2j46-stdenv-linux/setup: line 1596: pop_var_context: head of shell_variables not a function context
```

### Additional context

I did a git bisect for `python39Packages.python-prctl` and this is the commit where the breakage starts. Since it's a setuptools update that's probably not the root of the problem but I've @ed Martin below in case the type of error looks familiar.

```
1197a69edfd27c8fab83ea3d4408f1e13b504248 is the first bad commit
commit 1197a69edfd27c8fab83ea3d4408f1e13b504248
Author: Martin Weinelt <hexa@darmstadt.ccc.de>
Date:   Thu Jul 14 20:33:40 2022 +0200

    python3Packages.setuptools: 61.2.0 -> 63.2.0
```

### Notify maintainers

@catern 
@mweinelt

--- a/pkgs/development/python-modules/python-prctl/default.nix
+++ b/pkgs/development/python-modules/python-prctl/default.nix
@@ -4,6 +4,7 @@
   fetchPypi,
   libcap,
   pytestCheckHook,
+  distutils,
 }:
 
 buildPythonPackage rec {
@@ -18,7 +19,17 @@ buildPythonPackage rec {
 
   buildInputs = [ libcap ];
 
-  nativeCheckInputs = [ pytestCheckHook ];
+  nativeCheckInputs = [
+    distutils
+    pytestCheckHook
+  ];
+
+  postPatch = ''
+    substituteInPlace test_prctl.py \
+      --replace-fail \
+        'sys.version[0:3]' \
+        '"cpython-%d%d" % (sys.version_info.major, sys.version_info.minor)'
+  '';
 
   disabledTests = [
     # Intel MPX support was removed in GCC 9.1 & Linux kernel 5.6

